dm/syncer/syncer.go

```diff
@@ -1244,6 +1247,17 @@ func (s *Syncer) afterFlushCheckpoint(task *checkpointFlushTask) error {
 	s.lastCheckpointFlushedTime = now
 
 	s.logAndClearFilteredStatistics()
+
+	if s.cliArgs != nil && s.cliArgs.StartTime != "" {
+		clone := *s.cliArgs
+		clone.StartTime = ""
+		err2 := ha.PutTaskCliArgs(s.cli, s.cfg.Name, []string{s.cfg.SourceID}, clone)
+		if err2 != nil {
+			s.tctx.L().Error("failed to clean start-time in task cli args", zap.Error(err2))
```

Comment:
if user stop/start the task after this error, does task always tring to start sync from `s.cliArgs.StartTime`?

i can't find a better way to handle this scenario, but at least we should add sourceID/taskName in the error log

Reply:
Yes, and I think this is expected?

logger of syncer already have labels.